contracts/data-storage/andromeda-form/src/execute.rs

```diff
@@ -0,0 +1,418 @@
+use andromeda_data_storage::form::{ExecuteMsg, SubmissionInfo};
+use andromeda_modules::schema::{QueryMsg as SchemaQueryMsg, ValidateDataResponse};
+use andromeda_std::{
+    ado_contract::ADOContract,
+    amp::AndrAddr,
+    common::{
+        actions::call_action, context::ExecuteContext, encode_binary,
+        expiration::get_and_validate_start_time, Milliseconds,
+    },
+    error::ContractError,
+};
+use cosmwasm_std::{ensure, QueryRequest, Response, Uint64, WasmQuery};
+use cw_utils::{nonpayable, Expiration};
+
+use crate::{
+    contract::SUBMIT_FORM_ACTION,
+    state::{submissions, CONFIG, SCHEMA_ADO_ADDRESS, SUBMISSION_ID},
+};
+
+const MAX_LIMIT: u64 = 100u64;
+
+pub fn handle_execute(mut ctx: ExecuteContext, msg: ExecuteMsg) -> Result<Response, ContractError> {
+    let action_response = call_action(
+        &mut ctx.deps,
+        &ctx.info,
+        &ctx.env,
+        &ctx.amp_ctx,
+        msg.as_ref(),
+    )?;
+
+    let res = match msg {
+        ExecuteMsg::SubmitForm { data } => execute_submit_form(ctx, data),
+        ExecuteMsg::DeleteSubmission {
+            submission_id,
+            wallet_address,
+        } => execute_delete_submission(ctx, submission_id, wallet_address),
+        ExecuteMsg::EditSubmission {
+            submission_id,
+            wallet_address,
+            data,
+        } => execute_edit_submission(ctx, submission_id, wallet_address, data),
+        ExecuteMsg::OpenForm {} => execute_open_form(ctx),
+        ExecuteMsg::CloseForm {} => execute_close_form(ctx),
+        _ => ADOContract::default().execute(ctx, msg),
+    }?;
+
+    Ok(res
+        .add_submessages(action_response.messages)
+        .add_attributes(action_response.attributes)
+        .add_events(action_response.events))
+}
+
+pub fn execute_submit_form(
+    mut ctx: ExecuteContext,
+    data: String,
+) -> Result<Response, ContractError> {
+    nonpayable(&ctx.info)?;
+    let sender = ctx.info.sender;
+    ADOContract::default().is_permissioned(
+        ctx.deps.branch(),
+        ctx.env.clone(),
+        SUBMIT_FORM_ACTION,
+        sender.clone(),
+    )?;
+
+    let (expiration, _) = get_and_validate_start_time(&ctx.env, None)?;
+    let current_time = milliseconds_from_expiration(expiration)?;
```

Comment:
Wouldn't it be simpler to just extract current_time directly? Like     let current_time = Milliseconds::from_nanos(env.block.time.nanos()).milliseconds();
May not be the exact code but that's the idea. Also, we could delegate fetching the current time to `validate_form_is_opened`